test/puppetlabs/trapperkeeper/services/metrics/metrics_service_test.clj

```diff
@@ -230,11 +230,26 @@
     (with-app-with-config
       app
       (conj services authorization-service/authorization-service)
-      (merge metrics-service-config auth-config ssl-webserver-config)
+      ;; This test requires authentication so we need to override the require-auth
+      ;; option that is set to false in metrics-service-config
+      (-> (merge metrics-service-config auth-config ssl-webserver-config)
+          (assoc-in [:metrics :metrics-webservice :require-auth] true))
       (let [resp (http-client/get "https://localhost:8180/metrics/v2/list" ssl-opts)]
         (is (= 200 (:status resp))))
       (let [resp (http-client/get "https://localhost:8180/metrics/v2" ssl-opts)]
-        (is (= 403 (:status resp)))))))
+        (is (= 403 (:status resp))))))
+  (testing "exception occurs when require-auth option is true but auth service was not found"
+    ;; This test is like the previous, except the auth service is not listed as a dependency
+    (is (thrown-with-msg? ExceptionInfo
+                          #"authorization service was not found"
+                          (with-app-with-config
+                            app
+                            ;; The auth service is purposefully left out here
+                            services
+                            (-> (merge metrics-service-config auth-config ssl-webserver-config)
+                              (assoc-in [:metrics :metrics-webservice :require-auth] true)))))))
```

Comment:
Since you do this twice, might be good to put it in a `let` at the start of the test.